### PR TITLE
Set weekly availabilities

### PR DESCRIPTION
**Pull request by kevinl120**

### CODE REVIEW

**Comment by nova-admin** [comment]
#### 📝Changed routes:


| [**…/\[slug\]/availabilities**](https://risc-qq9s3mfut-novaforgood.vercel.app/program/[slug]/availabilities) | [**/test/testcalendar**](https://risc-qq9s3mfut-novaforgood.vercel.app/test/testcalendar) |
|:-:|:-:|
|<a href="https://risc-qq9s3mfut-novaforgood.vercel.app/program/[slug]/availabilities">Screenshot of /program/[slug]/availabilities</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/K3yADJcRrnU6snXS-TJPQ2vflDqj1a5MxJ0jiQI_96X2i1e1n4TyPiJ3uLlDRwb1F-GMkt3LrgNmFFrbsrATqA.png">(view full size)</a> |<a href="https://risc-qq9s3mfut-novaforgood.vercel.app/test/testcalendar">Screenshot of /test/testcalendar</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/OgKeD9fwh7IZSEeSan8TNxgkE2UgkgxRCKsAC_k8NqqtiZ2JB2vPbIahweJsaw91MempTkHAVleHXaNcHfzWkA.png">(view full size)</a> |







Commit 5d976c25b2a3014bd6d39c0dd105601b82454e04 (https://risc-qq9s3mfut-novaforgood.vercel.app).

File: codegen.yml [hunk 1]
```diff
@@ -3,12 +3,18 @@ schema: ${NEXT_PUBLIC_API_URL}
 documents: "src/graphql/**/*.graphql"
 generates:
   src/generated/graphql.tsx:
-    config:
-      withRefetchFn: true
     plugins:
+      - add:
+          content: "export type DateInterval = { startTime: Date, endTime: Date }"
```
[the remark this answers is not in the document]
Comment (legitmaxwu) [reply codegen.yml:8]: I get that the generated file needs this type but maybe we could add an import path somehow, or configure a globals.ts file so it doesn't have to be imported

File: src/components/atomic/Select.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Listbox, Transition } from "@headlessui/react";
-import _ from "lodash";
+import { isEqual } from "date-fns";
+import {} from "lodash";
```
Comment (legitmaxwu) [inline src/components/atomic/Select.tsx:3]: Whats's this line for?

File: src/components/Availabilities/SetWeeklyAvailabilitiesCard.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+              startTime: availabilitiesForDate[i].endTime,
+              endTime: addMinutes(availabilitiesForDate[i].endTime, 30),
+            };
+          }
+        }
+      }
+    }
+    if (!newAvailability) {
+      // Entire day is already available
+      return;
```
Comment (legitmaxwu) [inline src/components/Availabilities/SetWeeklyAvailabilitiesCard.tsx:228]: Throw/set an error indicator?

**Review by legitmaxwu — CHANGES_REQUESTED** [review]

**Review by legitmaxwu — APPROVED** [review]

**Pull request merged**